File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/Location.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class Location : DomainResource, IResource
{
    public string? Description { get; set; }
    public Base.Address? Address { get; set; }
    public Base.ResourceReference? ManagingOrganization { get; set; }
    public string? Name { get; set; }
    public string? Mode { get; set; }
    public Base.CodeableConcept[]? Type { get; set; }
    public string[]? Alias { get; set; }
    public string? Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Location_HoursOfOperation[]? HoursOfOperation { get; set; }
    public string? AvailabilityExceptions { get; set; }
    public Location_Position? Position { get; set; }
    public Base.ContactPoint[]? Telecom { get; set; }
    public Base.Coding? OperationalStatus { get; set; }
    public Base.ResourceReference? PartOf { get; set; }
    public Base.CodeableConcept? PhysicalType { get; set; }
    public Base.ResourceReference[]? Endpoint { get; set; }

    public class Location_HoursOfOperation : BackboneElement
    {
        public bool? AllDay { get; set; }
        public string[]? DaysOfWeek { get; set; }
        public string? ClosingTime { get; set; }
        public string? OpeningTime { get; set; }
    }

    public class Location_Position : BackboneElement
    {
        public string? Altitude { get; set; }
        public required string Latitude { get; set; }
        public required string Longitude { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Location.py
from typing import Optional, List
from base import Address
from base import CodeableConcept
from base import Coding
from base import ContactPoint
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class Location_HoursOfOperation(BackboneElement):
    all_day: Optional[bool] = None
    days_of_week: Optional[List[str]] = None
    closing_time: Optional[str] = None
    opening_time: Optional[str] = None

class Location_Position(BackboneElement):
    altitude: Optional[float] = None
    latitude: float
    longitude: float

class Location(DomainResource):
    description: Optional[str] = None
    address: Optional[Address] = None
    managing_organization: Optional[Reference] = None
    name: Optional[str] = None
    mode: Optional[str] = None
    type: Optional[List[CodeableConcept]] = None
    alias: Optional[List[str]] = None
    status: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    hours_of_operation: Optional[List[Location_HoursOfOperation]] = None
    availability_exceptions: Optional[str] = None
    position: Optional[Location_Position] = None
    telecom: Optional[List[ContactPoint]] = None
    operational_status: Optional[Coding] = None
    part_of: Optional[Reference] = None
    physical_type: Optional[CodeableConcept] = None
    endpoint: Optional[List[Reference]] = None